app/battle_i.py

```diff
@@ -0,0 +1,37 @@
+from typing import Dict
+from app.knight_config import Knight_Config
+
+
+def apply_effects(character: Knight_Config) -> None:
+    character["protection"] = sum(
+        [a["protection"] for a in character["armour"]]

```

Comment:
Use generator expression instead of list comprehension inside of `sum`. Besides that, do not use single-letter variable naming.